thehive/app/org/thp/thehive/services/PatternSrv.scala

```diff
@@ -0,0 +1,81 @@
+package org.thp.thehive.services
+
+import org.apache.tinkerpop.gremlin.structure.Graph
+import org.thp.scalligraph.EntityIdOrName
+import org.thp.scalligraph.auth.AuthContext
+import org.thp.scalligraph.models.{Database, Entity}
+import org.thp.scalligraph.services._
+import org.thp.scalligraph.traversal.TraversalOps._
+import org.thp.scalligraph.traversal.{Converter, Traversal}
+import org.thp.thehive.models._
+import org.thp.thehive.services.CaseOps._
+import org.thp.thehive.services.PatternOps._
+import org.thp.thehive.services.ProcedureOps._
+
+import java.util.{Map => JMap}
+import javax.inject.{Inject, Named, Singleton}
+import scala.util.{Success, Try}
+
+@Singleton
+class PatternSrv @Inject() (
+    auditSrv: AuditSrv,
+    caseSrv: CaseSrv,
+    organisationSrv: OrganisationSrv
+)(implicit
+    @Named("with-thehive-schema") db: Database
+) extends VertexSrv[Pattern] {
+  val patternPatternSrv = new EdgeSrv[PatternPattern, Pattern, Pattern]
+
+  def parentExists(child: Pattern with Entity, parent: Pattern with Entity)(implicit graph: Graph): Boolean =
+    child._id == parent._id || get(child).parent.getEntity(parent).exists
```

Comment:
When child is parent ?

Reply:
I badly named this function, is `cannotBeParent` better ?